modules/enterprise/server/jar/src/main/java/org/rhq/enterprise/server/content/ContentManagerBean.java

```diff
@@ -671,47 +650,56 @@ public void removeHistoryDeploymentsBits(Subject user){
         for(Package pkg: packages){
             purgePackageBits(pkg.getId());
         }
-        deleteOrphanedBlobs();
+        Query deleteBitsQuery = entityManager.createNamedQuery(PackageBits.DELETE_IF_NO_PACKAGE_VERSION);
+        deleteBitsQuery.executeUpdate();
     }
 
-    private void deleteOrphanedBlobs(){
-        // Delete orphanate bits
-        if(DatabaseTypeFactory.isPostgres(DatabaseTypeFactory.getDefaultDatabaseType())) {
-            try {
-                Connection conn = dataSource.getConnection();
-                PreparedStatement ps =
-                        conn.prepareStatement("SELECT BITS FROM " + PackageBits.TABLE_NAME + " WHERE ID NOT IN ( " +
-                                "SELECT PACKAGE_BITS_ID FROM RHQ_PACKAGE_VERSION WHERE PACKAGE_BITS_ID IS NOT NULL )");
-                ResultSet rs = ps.executeQuery();
+    public void removeHistoryDeploymentsBits(Subject user){
```

Comment:
Since is now on the purgeJob, it might be worth to remove this public API (one thing less to document).